### PR TITLE
Improve convergence behavior with groupnorm

### PR DESCRIPTION
**Pull request by michaelmckinsey1**
Replaced BatchNorm3d in the 3D U-Net with GroupNorm, and made the number of groups configurable via `group_norm_groups`. This improves training stability for low batch sizes.

## 1 minibatch size
batchnorm
image

groupnorm
image

- [x] need to check if this is worse at higher batch sizes than batch norm

groupnorm does converge in up to 2x less epochs, even at higher batch sizes
## 8 minibatch size
batchnorm
image

groupnorm
image

- ~should we give an option for either group norm or batch norm?~
   - I think this is only necessary if groupnorm is significantly worse at higher batch sizes, which it isn't.

### CODE REVIEW

**Review by PatrickRMiles — APPROVED** [review]
Looks good!
